src/main/java/org/interledger/Fulfillment.java

```diff
@@ -0,0 +1,106 @@
+package org.interledger;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+/**
+ * The fulfillment of a {@link Condition}.
+ */
+public class Fulfillment {
+
+  private byte[] preimage;
+  private Condition condition;
+
+  /**
+   * Create a new Fulfillment from the 32 byte pre-image that is hashed to get the condition.
+   * 
+   * @param preimage A 32 byte octet string
```

Comment:
Might be worth clarifying what a `pre-image` is. Without being familiar with either crypto in general, or the crypto-conditions spec, this term may not make sense.